File: java/Adapter/src/adapter/GUIAdapter.java
package adapter;

public class GUIAdapter {
	WindowsGUI winGUI;
	LinuxGUI linuxGUI;
	MacGUI macGUI;

	boolean isWindows;
	boolean isLinux;

	public GUIAdapter(String operatingSystem) {
		if (operatingSystem.contains("Windows")) {
			winGUI = new WindowsGUI();
			isWindows = true;
		} else if (operatingSystem.contains("Linux")) {
			linuxGUI = new LinuxGUI();
			isLinux = true;
		} else {
			macGUI = new MacGUI();
		}

	}

	public void display() {
		if (isWindows) {
			winGUI.________(); // > 1
		} else if (isLinux) {
			linuxGUI._______(); // > 2
		} else {
			macGUI.______(); // > 3
		}
	}
}

File: ruby/Adapter/gui_adapter.rb
require_relative 'windows_gui'
require_relative 'mac_gui'
require_relative 'linux_gui'

class GUIAdapter
  def initialize(operating_system)
   @os = operating_system

   if operating_system.include? "windows"
     @windows = WindowsGUI.new
   elsif operating_system.include? "linux"
     @linux = LinuxGUI.new
   elsif operating_system.include? "Mac"
     @mac = MacGUI.new
   end
  end

  def display
    if @os.include? "windows"
      @windows._____ # >> 1
    elsif @os.include? "linux"
      @linux._____ # >> 2
    elsif @os.include? "Mac"
      @mac._______ # >> 3
    end
  end

end